services/src/main/java/org/fao/geonet/api/records/attachments/AttachmentsApi.java

```diff
@@ -241,47 +249,62 @@ public MetadataResource putResourceFromURL(
         ServiceContext context = ApiUtils.createServiceContext(request);
-        try (Store.ResourceHolder file = store.getResource(context, metadataUuid, resourceId, approved)) {
-
-            ApiUtils.canViewRecord(metadataUuid, request);
-
-            MultiValueMap<String, String> headers = new HttpHeaders();
-            headers.add("Content-Disposition", "inline; filename=\"" + file.getMetadata().getFilename() + "\"");
-            headers.add("Cache-Control", "no-cache");
-            String contentType = getFileContentType(file.getPath());
-            headers.add("Content-Type", contentType);
-
-            if (contentType.startsWith("image/") && size != null) {
-                if (size >= MIN_IMAGE_SIZE && size <= MAX_IMAGE_SIZE) {
-                    BufferedImage image = ImageIO.read(file.getPath().toFile());
-                    BufferedImage resized = ImageUtil.resize(image, size);
-                    ByteArrayOutputStream output = new ByteArrayOutputStream();
-                    ImageIO.write(resized, "png", output);
-                    output.flush();
-                    byte[] imagesB = output.toByteArray();
-                    output.close();
-                    return new HttpEntity<>(imagesB, headers);
+
+        ApiUtils.canViewRecord(metadataUuid, request);
+
+        ServletOutputStream out = response.getOutputStream();
+
+        if (store instanceof FMEStore
+        || (store instanceof ResourceLoggerStore && ((ResourceLoggerStore) store).getDecoratedStore() instanceof FMEStore)) {
+            response.setHeader("Content-Disposition", "inline; filename=\"" + resourceId + "\"");
+            response.setHeader("Cache-Control", "no-cache");
+
+            store.streamResource(context, metadataUuid, resourceId, approved, out);
+        } else {
+            try (Store.ResourceHolder file = store.getResource(context, metadataUuid, resourceId, approved)) {
+                response.setHeader("Content-Disposition", "inline; filename=\"" + file.getMetadata().getFilename() + "\"");
+                response.setHeader("Cache-Control", "no-cache");
+
+                Path path = file.getPath();
+                String contentType = getFileContentType(path);
+                response.setHeader("Content-Type", contentType);
+
+                if (contentType.startsWith("image/") && size != null) {
+                    if (size >= MIN_IMAGE_SIZE && size <= MAX_IMAGE_SIZE) {
+                        BufferedImage image = ImageIO.read(path.toFile());
+                        BufferedImage resized = ImageUtil.resize(image, size);
+                        ByteArrayOutputStream output = new ByteArrayOutputStream();
+                        ImageIO.write(resized, "png", output);
+                        output.flush();
+                        byte[] imagesB = output.toByteArray();
+                        output.close();
+                        out.write(imagesB);
```

Comment:
It would be better to stream the file to the servlet's output stream instead of loading it in memory as `byte[]`. 
The temporary `output` file with the resized image should be deleted after used.